### DISCUSSION TITLE
Get procedure path from the TRPCProxyClient

### DISCUSSION THREAD

**Discussion by lkwr** [trpc/trpc#3909] [Q&A]
Hello, lets imagine I have a `TRPCProxyClient` like this:

```typescript
const trpc = createTRPCProxyClient<Router>({
  links: [...]
});
```

and the Server has the following router:

```typescript
const router = t.router({
  demo: t.router({
    hello: t.procedure.query(() => "Hello"),
  }),
  test: t.router({
    foo: t.procedure.query(() => "Foo"),
    bar: t.procedure.query(() => "Bar"),
  })
})
```

Now what I want is this on the client-side:

```typescript
const myProcedure = trpc.test.foo;
const myProcedurePath: string = someMagicFunction(myProcedure);

// myProcedurePath should be something like this: "test.foo"
```

Is there any way to get something like this above? I need something like this as a unique cache key for an internal helper function in my project. I know I can write a cache key manually everytime but I want to make this automatic.

**Comment by lilienblum** [comment]
It seems there's no option unless you make your own wrapper
